cogip/models/parameter.py

```diff
@@ -0,0 +1,316 @@
+    type: Literal["int32"] = "int32"
+    content: Annotated[int, Strict]
+
+
+class ParameterUInt32(ParameterBase):
+    """Unsigned 32-bit integer parameter value."""
+
+    type: Literal["uint32"] = "uint32"
+    content: Annotated[NonNegativeInt, Strict]
+
+
+class ParameterInt64(ParameterBase):
+    """Signed 64-bit integer parameter value."""
+
+    type: Literal["int64"] = "int64"
+    content: Annotated[int, Strict]
+
+
+class ParameterUInt64(ParameterBase):
+    """Unsigned 64-bit integer parameter value."""
+
+    type: Literal["uint64"] = "uint64"
+    content: Annotated[NonNegativeInt, Strict]
+
+
+class ParameterBool(ParameterBase):
+    """Boolean parameter value."""
+
+    type: Literal["bool"] = "bool"
+    content: Annotated[bool, Strict]
+
+
+# Discriminated union of all parameter value types
+ParameterValueType = (
+    ParameterFloat
+    | ParameterDouble
+    | ParameterInt32
+    | ParameterUInt32
+    | ParameterInt64
+    | ParameterUInt64
+    | ParameterBool
+)
+
+
+class Parameter(BaseModel):
+    """Parameter model with discriminated union for type-safe values.
+
+    Attributes:
+        name: The parameter name
+        value: The parameter value with its type
+    """
+
+    model_config = {"validate_assignment": True}
+
+    name: str
+    value: Annotated[ParameterValueType, Field(discriminator="type")]
+
+    def __hash__(self):
+        return fnv1a_hash(self.name)
+
+    def get_value(self) -> float | int | bool:
+        """Get the parameter content value.
+
+        Returns:
+            The actual content value (float, int, or bool)
+        """
+        return self.value.content
+
+    def set_value(self, content: float | int | bool) -> None:
+        """Set the parameter content value.
+
+        Args:
+            content: The new content value to set
+
+        Note:
+            The type of the parameter remains unchanged. The content must be
+            compatible with the existing parameter type.
+        """
+        self.value.content = content
+
+    def pb_copy(self, message: PB_ParameterSetRequest | PB_ParameterGetRequest) -> None:
+        """Copy values to Protobuf message"""
+        message.key_hash = hash(self)
+
+        if isinstance(message, PB_ParameterSetRequest):
+            match self.value.type:
+                case "float":
+                    message.value.float_value = self.value.content
+                case "double":
+                    message.value.double_value = self.value.content
+                case "int32":
+                    message.value.int32_value = self.value.content
+                case "uint32":
+                    message.value.uint32_value = self.value.content
+                case "int64":
+                    message.value.int64_value = self.value.content
+                case "uint64":
+                    message.value.uint64_value = self.value.content
+                case "bool":
+                    message.value.bool_value = self.value.content
```

Comment:
You could maybe shorten this statement using just: `message.value.setattr(f"{self.value.type}_value", self.value.content)`.